### DISCUSSION TITLE
Extracting values

### DISCUSSION THREAD

**Discussion by gishan12** [pybamm-team/PyBaMM#2060] [Q&A] [answered]
Hi, in the full_diffusion.py file, these are lines 70-72:
''
N_e_diffusion = -tor * param.D_e(c_e, T) * pybamm.grad(c_e)
N_e_migration = param.C_e * param.t_plus(c_e, T) * i_e / param.gamma_e
N_e_convection = param.C_e * c_e * v_box
"

I just want to be able to see the numerical value of param.C_e. Could someone please tell me which pybamm expression could allow me to do that (or whether it is even possible). Please let me know, it would mean a lot. Thank you!

**Comment by valentinsulzer** [comment, marked as answer]
This will print all the (non-function) parameters for a given parameter set to `output_file`:
```python
import pybamm
parameters = pybamm.LithiumIonParameters()
parameter_values = pybamm.ParameterValues("Chen2020")
output_file = "lithium_ion_parameters.txt"
parameter_values.print_parameters(parameters, output_file)
```

To view the values of `N_e_diffusion/migration/convection`, you would need to add them to the variables dictionary by editing the source code in `full_diffusion.py`

**Reply by gishan12** [reply]
Thank you so so much, trying it right now!

**Reply by gishan12** [reply]
it worked, thank you so much!
